### PR TITLE
booting-qemu-image: Add runqemu details and build instructions

### PR DESCRIPTION
**Pull request by fhajredini**
Signed-off-by: Fisnik Hajredini <fhajredini@luxoft.com>

### CODE REVIEW

**Review by JEderonn — APPROVED** [review]

File: docs/articles/baseplatform/booting-qemu-image.rst
Comment (sashko) [inline docs/articles/baseplatform/booting-qemu-image.rst:13]: why did you remove the header?

File: docs/articles/baseplatform/booting-qemu-image.rst
Comment (JEderonn) [inline docs/articles/baseplatform/booting-qemu-image.rst]: like what flexibility, just state some examples

File: docs/articles/baseplatform/booting-qemu-image.rst
Comment (fhajredini) [reply docs/articles/baseplatform/booting-qemu-image.rst:13]: its on line 63 now..

Comment (fhajredini) [reply docs/articles/baseplatform/booting-qemu-image.rst:13]: or do you mean something else?

**Comment by fhajredini** [comment]
I will merge this by the end of the day unless there are further comments!

File: docs/articles/baseplatform/booting-qemu-image.rst
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: "a part" -> "part" (in two occurrences)

"straight forward" -> "straightforward"

Also, please break long lines at 80 chars.

File: docs/articles/baseplatform/booting-qemu-image.rst
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: to ssh -> to run ``ssh`` (move the ssh inside the monospace font)

File: docs/articles/baseplatform/booting-qemu-image.rst
Comment (mardy) [inline docs/articles/baseplatform/booting-qemu-image.rst]: no need to abbreviate dev and prod

**Review by mardy — CHANGES_REQUESTED** [review]
Just some nitpicking :-)

File: docs/articles/baseplatform/booting-qemu-image.rst
Comment (fhajredini) [reply docs/articles/baseplatform/booting-qemu-image.rst]: I did that change, but how about removing the development and production entirely, since for QEMU we only provide the minimal and minimal-dev images. Also neither one of them is production..

Comment (mardy) [reply docs/articles/baseplatform/booting-qemu-image.rst]: Good point! :-)

Comment (martin-ejdestig) [reply docs/articles/baseplatform/booting-qemu-image.rst]: So, I do not really understand why qemu_launcher is needed. As far as I know QEMU uses KVM if it is available. (If the `runqemu` script does not use KVM by default if it is available, one just has to append `kvm` to the command line, it looks like.)

And `qemu_launcher` does not use QEMU? I don't know... the more I look at it the more confused I get. Commit message nor #97 contains any information as to why it is needed and why it should be used instead of `runqemu`. :(

(Asking this question here since I am not sure adding a section for it makes sense... if it is not needed. :)

Comment (martin-ejdestig) [reply docs/articles/baseplatform/booting-qemu-image.rst]: If the `runqemu` script does not use KVM by default if it is available, one just has to append `kvm` to the command line, it looks like.

Comment (martin-ejdestig) [reply docs/articles/baseplatform/booting-qemu-image.rst]: And `qemu_launcher` does not use QEMU? I don't know... the more I look at it the more confused I get.

Comment (martin-ejdestig) [reply docs/articles/baseplatform/booting-qemu-image.rst]: Commit message nor #97 contains any information as to why it is needed and why it should be used instead of `runqemu`. :(

Comment (martin-ejdestig) [reply docs/articles/baseplatform/booting-qemu-image.rst]: Anyway, not blocking this to get merged. I am just confused and wonder if it really makes sense to have it at all... and if we can't explain to ourselves why it is needed, how can we do so to someone reading this section. :)

Comment (fhajredini) [reply docs/articles/baseplatform/booting-qemu-image.rst]: >As far as I know QEMU uses KVM if it is available.

That happens when u start QEMU with `runqemu`. On the other hand, if you want to use qemu_launcher, you must have KVM. 

>  qemu_launcher does not use QEMU?

It does.. I dont get it at what point did this confuse you?

> Commit message nor #97 contains any information 

Are you sure u meant about #97? It doesnt seem relevant to this PR. 
However, my understanding of why we should use qemu_launcher is that it has been developed internally.

Comment (fhajredini) [reply docs/articles/baseplatform/booting-qemu-image.rst]: Generally speaking tho, i was tempted to remove any information about qemu_launcher, but after some discussion it was decided to only add the information about runqemu and leave the topic of qemu_launcher for advanced users.

Comment (martin-ejdestig) [reply docs/articles/baseplatform/booting-qemu-image.rst]: > > As far as I know QEMU uses KVM if it is available.
> 
> That happens when u start QEMU with `runqemu`. On the other hand, if you want to use qemu_launcher, you must have KVM.
> 

Yes, so what is the point?

> > qemu_launcher does not use QEMU?
> 
> It does.. I dont get it at what point did this confuse you?

Nope, it does `sudo kvm ...` . "QEMU" is only used in help strings etc.

> > Commit message nor #97 contains any information
> 
> Are you sure u meant about #97? It doesnt seem relevant to this PR.

Sorry, the PR is in meta-pelux: https://github.com/Pelagicore/meta-pelux/pull/97 .

Comment (fhajredini) [reply docs/articles/baseplatform/booting-qemu-image.rst]: So you recommending to remove the section about qemu_launcher entirely? I dont really have a strong opinion on that and i would agree that it only confuses the user.

**Comment by fhajredini** [comment]
For now, im leaving the qemu_launcher instructions as no final decision has been taken.

**Pull request merged**
